packages/core/src/browser/widgets/widget.ts

```diff
@@ -254,9 +254,10 @@ export type EventListenerOrEventListenerObject<K extends keyof HTMLElementEventM
 export function addEventListener<K extends keyof HTMLElementEventMap>(
     element: HTMLElement, type: K, listener: EventListenerOrEventListenerObject<K>, useCapture?: boolean
 ): Disposable {
-    element.addEventListener(type, listener, useCapture);
+    // The DOM API typings don't seem to take a specific listener as described in this function.
+    element.addEventListener(type, listener as EventListenerOrEventListenerObject<any>, useCapture);
```

Comment:
Changing the type of `EventListenerOrEventListenerObject<K>` to `(this: HTMLElement, ev: HTMLElementEventMap[K]) => unknown` (derived from the DOM `element.addEventListener` typing) appears to fix this without introducing any new type errors.

Reply:
It doesn't show an error in the editor but it fails at build time, will leave as is for now.